### PR TITLE
Check widget: Only respond to hover and tap over the checkbox or label

### LINKED ISSUE #4527: Check widget: hovering/tapping to the right of the label area should not activate widget
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

If the Check widget is wider than its label, tapping outside the label toggles it, which it should not. Related to #3528 

### How to reproduce

Fyne demo, Input tab

### Screenshots

_No response_

### Example code

fyne demo

### Fyne version

develop

### Go compiler version

go1.21.1

### Operating system and version

all

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by dweymouth**
### Description:

Fixes #4527, related to #4320 

Updates the Check widget to only respond to hover and tap events over either the checkbox or the label area of the widget. If the widget is wider or taller because it is placed in a container that maximizes its size, ignore events outside the "active" area of the widget.

The first commit of this PR just reordered the functions in `check.go` in a way that made more sense (resulting in an inflated line count diff relative to the actual code change). To review, you may wish to only look at the second commit.

Note: the behavior can be tested with fyne_demo. Before this change the check widget could be toggled by tapping far to the right, well beyond the end of the label. With the behavior, tapping/hovering in that area of the widget (which intuitively to the user is not part of the widget at all) does nothing

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

**Comment by Jacalz** [comment]
Did you mean to request a review? :)

**Comment by dweymouth** [comment]
Yes, this pr is ready for review :)

**Comment by Jacalz** [comment]
I mean given that you assigned me and Andrew to it? ;)

**Comment by dweymouth** [comment]
Just wasn't sure how you had notifications setup

**Comment by Jacalz** [comment]
Alright, cool. Absolutely no problem. Just though you might have wanted us to do something else than review :)

File: widget/check.go [hunk 1]
```diff
@@ -167,42 +93,65 @@ func (c *Check) Hide() {
...
+
+	oldHovered := c.hovered
+	if c.minSize.Width == 0 && c.minSize.Height == 0 {
+		c.hovered = true
+	} else if me.Position.X <= c.minSize.Width && me.Position.Y <= c.minSize.Height {
+		c.hovered = true
+	} else {
+		// mouse outside the active area of the widget
+		c.hovered = false
+	}
```
Comment (Jacalz) [inline widget/check.go:122]: You can simplify the code and make it branch-free like the code example below. I think it also is a bit more readable perhaps?
```go
	c.hovered = c.minSize.IsZero() || (me.Position.X <= c.minSize.Width && me.Position.Y <= c.minSize.Height)
```

**Review by Jacalz — CHANGES_REQUESTED** [review]
Thanks. Really nice change. More consistency with the `Radio` widget :+1: 
Just left one minor suggestion inline

File: widget/check.go (hunk 1 above)
Comment (dweymouth) [reply widget/check.go]: Nice one

**Review by Jacalz — APPROVED** [review]
Wonderful work. Thanks :)

**Pull request merged**